src/cargo/core/compiler/mod.rs

```diff
@@ -707,13 +708,24 @@ fn add_error_format_and_color(
         // to emit a message that cargo will intercept.
         json.push_str(",artifacts");
     }
+
     match cx.bcx.build_config.message_format {
         MessageFormat::Short | MessageFormat::Json { short: true, .. } => {
             json.push_str(",diagnostic-short");
         }
         _ => {}
     }
     cmd.arg(json);
+
+    if nightly_features_allowed() {
```

Comment:
As far as I can tell, this wasn't opt-in because if `accurate_err_width` returned `Some(_)` then the flag would be added without explicit instruction from the user. When `accurate_err_width` returned `None`, then the user would have to pass `-Z terminal-width` and that would be opting-in. 

So I've chosen to remove the `accurate_err_width` parts of this change - unless you want me to add a `-Z auto-terminal-width` boolean flag (or similar), any opting-in with `-Z terminal-width` will necessarily specify a width which make the `accurate_err_width` changes redundant.

Reply:
I think the idea is that `-Zterminal-width` (without a value) would have Cargo compute the width and pass it along (if possible), which would use the `accurate_stderr_width` functions.  I'm not sure if there's an actual use case for manually passing in the width?  Like, long-term, what is the expected behavior?  I would assume that if this is stabilized, there would be no options, and it would be automatic.

Reply:
Alright, I've made it so that `-Z terminal-width` can take no value or a usize.